apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts

```diff
@@ -0,0 +1,688 @@
+    if (!compare) compare = (idA ?? 0) - (idB ?? 0);
+    if (!compare) compare = rowA.number - rowB.number;
+    if (!compare) compare = valueA - valueB;
+    return compare;
+  }
+
+  function timeRemainingLimitSorter(
+    valueA: number,
+    valueB: number,
+    rowA: AssessmentInstanceRow,
+    rowB: AssessmentInstanceRow,
+  ) {
+    // Closed assessments are listed first, followed by time limits
+    // ascending, followed by open without a time limit
+    return Number(rowA.open) - Number(rowB.open) || (valueA ?? Infinity) - (valueB ?? Infinity);
+  }
+
+  function actionButtonFormatter(_value: string, row: AssessmentInstanceRow) {
+    const ai_id = row.assessment_instance_id;
+    const container = $('<div>');
+    const dropdown = $('<div class="dropdown">').appendTo(container);
+    $('<button type="button">')
+      .addClass('btn btn-secondary btn-xs dropdown-toggle')
+      .attr('data-toggle', 'dropdown')
+      .attr('aria-haspopup', 'true')
+      .attr('aria-expanded', 'false')
+      .attr('data-boundary', 'window')
+      .text('Action')
+      .appendTo(dropdown);
+    $('<div>')
+      .attr('id', `row${ai_id}PopoverClose`)
+      .attr('tabindex', 0)
+      .attr('data-toggle', 'popover')
+      .attr('title', 'Confirm close')
+      .attr(
+        'data-content',
+        `<form name="close-form" method="POST" onsubmit="popoverSubmitViaAjax(event, '#row${ai_id}PopoverClose')">
+                 <input type="hidden" name="__action" value="close">
+                 <input type="hidden" name="__csrf_token" value="${csrfToken}">
+                 <input type="hidden" name="assessment_instance_id" value="${ai_id}">
+                 <button type="button" class="btn btn-secondary" onclick="$('#row${ai_id}PopoverClose').popover('hide')">Cancel</button>
+                 <button type="submit" class="btn btn-danger">Grade and close</button>
+               </form>`,
+      )
+      .appendTo(dropdown);
+    $('<div>')
+      .attr('id', `row${ai_id}PopoverRegrade`)
+      .attr('tabindex', 0)
+      .attr('data-toggle', 'popover')
+      .attr('title', 'Confirm regrade')
+      .attr(
+        'data-content',
+        `<form name="regrade-form" method="POST">
+                 <input type="hidden" name="__action" value="regrade">
+                 <input type="hidden" name="__csrf_token" value="${csrfToken}">
+                 <input type="hidden" name="assessment_instance_id" value="${ai_id}">
+                 <button type="button" class="btn btn-secondary" onclick="$('#row${ai_id}PopoverRegrade').popover('hide')">Cancel</button>
+                 <button type="submit" class="btn btn-primary">Regrade</button>
+               </form>`,
+      )
+      .appendTo(dropdown);
+    const menu = $('<div>')
+      .addClass('dropdown-menu')
+      .attr('onclick', 'window.event.preventDefault()')
+      .appendTo(dropdown);
+    //<% if (authz_data.has_course_instance_permission_edit) { %>
```

Comment:
Are you accounting for this anywhere else? It seems like this may have been improperly converted from EJS.

Reply:
No, now accounting for this in [306ccbe](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/306ccbe9c0ebd4481747d7e7ddce945db6fff328).